File: func1/func.rb
require 'fdk'
require './models/user.rb'

def myhandler(context, input)
  STDERR.puts "call_id: " + context.call_id
  name = "World"
  nin = input['name']
  if nin && nin != ""
    name = nin
  end
  u = User.new(1, 'banana', 'banana@eurus.cn', '110')
  STDERR.puts u.deserialize(u.serialize)
  
  return {message: u.serialize}
end

FDK.handle(:myhandler)

File: func2/func.js
var fdk = require('@fnproject/fdk');

var messages = require('./messages/user_message_pb');

fdk.handle(function(input) {

    var name = 'World';
    if (input.name) {
        name = input.name;
    }
    var message = new messages.UserMessage();

    message.setId(1);
    message.setName(name);
    message.setEmail('banana@eurus.cn');
    message.setMobile('110')

    var bytes = message.serializeBinary();
    var dbytes = messages.UserMessage.deserializeBinary(bytes);
    response = {
        'message': bytes,
        'origin': dbytes
    }

    return response
})

